### DISCUSSION TITLE
Set Ids manually?

### DISCUSSION THREAD

**Discussion by dominiks** [bitbrain/pandora#145] [General]
The ids for entities and categories are currently auto generated numbers. Seeing as they are strings internally, would it be possible to be able to set them manually? I'd feel a bit more comfortable using "item_torch" instead of a random number when creating entities or writing serialized data or loading it. I know there's the auto generated id-class but that only works for the code but not for serialization.

**Comment by bitbrain** [comment]
This could indeed work. However, to achieve this, we'd need better logic to protect against duplicate ids, e.g. if you are in control of the ids, creating an entity with an already existing id could have catastrophic consequences.
